docs/DeveloperGuide.md

```diff
@@ -1,34 +1,503 @@
 # Developer Guide
 
-## Design & implementation
+## Introduction
+This section provides an introduction to the developer guide for *25HoursADay*.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Introduction to *25HoursADay*
+*25HoursADay* is a scheduling app optimized for users via a convenient yet powerful Command Line Interface (CLI) while still having the benefits of a Graphical User Interface (GUI). If you can type fast, *25HoursADay* can schedule your tasks and events faster than traditional GUI apps.
 
+### Purpose and scope
+The purpose of this developer guide is to describe the architecture and software design decisions for the task scheduling application *25HoursADay*. This developer guide covers an overview of the program architecture, the logical view of major components, and the mechanism of the functions.
+
+The intended audience of this guide are the developers and software testers of *25HoursADay*.
+
+### Developer guide organization
+The table below shows the content of the developer guide.
+
+|Section| Purpose | 
+|--------|----------|
+|[Section 2: Setting up](#setting-up)|To provide instructions on how to download and set up *25HoursADay* on your computer.|
+|[Section 3: Design](#design)|To explain the architecture of *25HoursADay*,  and describe the major components of *25HoursADay*, the roles of major components, as well as their organization and interaction of major components.|
+|[Section 4: Implementation](#implementation)|To explain the mechanism of functions added to *25HoursADay* and the features added to *25HoursADay*.|
+|[Section 5: Documentation](#documentation)|To represent documents describing the system and its parts.|
+
+## Setting Up & Getting started
```

Comment:
Perhaps can label the main headings with the section numbers for better readability.